### PR TITLE
Enable prow jobs for kubernetes/federation

### PR DESCRIPTION
**Pull request by marun**
/area prow
/assign cjwagner

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -42,6 +43,92 @@ presubmits:
...
+      - image: gcr.io/k8s-testimages/bazelbuild:v20171013-53473ec5-0.6.1
+        args:
+        - "--clean"
+        - "--git-cache=/root/.cache/git"
+        - "--job=$(JOB_NAME)"
+        - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
+        - "--service-account=/etc/service-account/service-account.json"
+        - "--upload=gs://kubernetes-jenkins/pr-logs"
+        - "--" # end bootstrap args, scenario args below
+        - "--test=//... -//build/... -//vendor/..."
```
Comment (cjwagner) [inline prow/config.yaml]: I think this will fail until there are BUILD files under these directories.

**Comment by cjwagner** [comment]
A mungegithub instance running the approval-handler and OWENRS files need to be set up before Tide will find anything with the new query (nothing will have the `approved` label until then).

**Comment by marun** [comment]
@cjwagner Does the OWNERS need to be merged to the repo or can it be included in PR that will seed the repo with code from k/k?

File: prow/config.yaml (hunk 1 above)
Comment (marun) [reply prow/config.yaml]: There will be build files under those directories.

Comment (BenTheElder) [reply prow/config.yaml]: This will definitely fail until there is a WORKSPACE.

**Comment by cjwagner** [comment]
> Does the OWNERS need to be merged to the repo or can it be included in PR that will seed the repo with code from k/k?

I'm not sure I understand the distinction you are making. Are you asking if the OWNERS file needs to be added before the seed PR?
In order for automatic merging to work right away there needs to be a root level OWNERS file before the approval-handler is enabled, otherwise no one will be able to approve changes to the root level. 

I assume you have permission to use the merge button in the new repo though so you can just use that to add the root level OWNERS file and merge the seed PR in any order.

File: prow/config.yaml [hunk 2]
```diff
@@ -42,6 +43,92 @@ presubmits:
     context: pull-charts-e2e
     rerun_command: "/test pull-charts-e2e"
     trigger: "(?m)^/test( all| pull-charts-e2e),?(\\s+|$)"
+  kubernetes/federation:
```
Comment (spxtr) [inline prow/config.yaml]: I recommend splitting this into two PRs: one to add this job and then another to enable tide.

Comment (marun) [reply prow/config.yaml]: Done.

File: prow/config.yaml (hunk 1 above)
Comment (marun) [reply prow/config.yaml]: The initial PR for federation is intended to include everything necessary to get these jobs passing.

File: prow/config.yaml (hunk 2 above)
Comment (marun) [reply prow/config.yaml]: tide PR is: https://github.com/kubernetes/test-infra/pull/5167

**Pull request merged**